central/complianceoperator/v2/compliancemanager/manager_impl.go

```diff
@@ -95,6 +95,8 @@ func (m *managerImpl) ProcessScanRequest(ctx context.Context, scanRequest *stora
 		return nil, errors.Errorf("Compliance is disabled. Cannot process scan request: %q", scanRequest.GetScanConfigName())
 	}
 
+	createScanRequest := false
```

Comment:
nit: move this line closer to its use.
And you can use 
`createScanRequest := scanRequest.Id == ""`